programs/ziti-edge-tunnel/instance.c

```diff
@@ -313,11 +314,11 @@ static void setTunnelServiceAddress(tunnel_service *tnl_svc, ziti_service *servi
         for(int port_idx = 0; cfg_v1.port_ranges[port_idx]; port_idx++) {
             tnl_port_range_arr[port_idx] = getTunnelPortRange(cfg_v1.port_ranges[port_idx]);
         }
-        if (cfg_v1.addresses) free(cfg_v1.addresses);
-        if (cfg_v1.port_ranges) free(cfg_v1.port_ranges);
+        cfg_v1.protocols = NULL;
+        free_ziti_intercept_cfg_v1(&cfg_v1);
     }  else if ((cfg_json = ziti_service_get_raw_config(service, CFG_ZITI_TUNNELER_CLIENT_V1)) != NULL) {
         ZITI_LOG(TRACE, "ziti-tunneler-client.v1: %s", cfg_json);
-        ziti_client_cfg_v1 zt_client_cfg_v1;
+        ziti_client_cfg_v1 zt_client_cfg_v1 = {0};
```

Comment:
initializing with `{ 0 }` is not necessary, since `parse_obj` uses `memset` to zero the object before parsing.